### PR TITLE
Subgraphs v3

### PR DESCRIPTION
**Pull request by fritzschoff**

### CODE REVIEW

File: v3/subgraphs/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+type PoolCreated @entity {
```
Comment (noahlitvin) [inline v3/subgraphs/schema.graphql:1]: rename to `Pool`

File: v3/subgraphs/schema.graphql [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  id: ID!
+  block: BigInt!
+  owner: Bytes!
+  timestamp: BigInt!
```
Comment (noahlitvin) [inline v3/subgraphs/schema.graphql:5]: rename to `created_at`

File: v3/subgraphs/schema.graphql [hunk 3]
```diff
@@ -0,0 +1,6 @@
+type PoolCreated @entity {
+  id: ID!
+  block: BigInt!
+  owner: Bytes!
```
Comment (noahlitvin) [inline v3/subgraphs/schema.graphql:4]: Is there an Address type or is this supposed to be Bytes? I forget...

**Review by noahlitvin — CHANGES_REQUESTED** [review]

File: v3/subgraphs/schema.graphql (hunk 3 above)
Comment (fritzschoff) [reply v3/subgraphs/schema.graphql:4]: Bildschirmfoto 2022-11-07 um 14 34 59
dont't think so. Maybe String?

Comment (fritzschoff) [reply v3/subgraphs/schema.graphql:4]: Bildschirmfoto 2022-11-07 um 14 38 53
Bytes is suggested by the CLI

Comment (fritzschoff) [reply v3/subgraphs/schema.graphql:4]: Will go with Bytes

File: .eslintignore [hunk 4]
```diff
@@ -12,4 +12,5 @@ packages/queries/subgraphs
 **/storybook-static/
 **/.nyc_output
 **/junit.xml
+**/generated
```
[the remark this answers is not in the document]
Comment (fritzschoff) [reply .eslintignore:15]: and how do you handle the complaints from es lint when it doesn't fit into the rules?

File: v3/subgraphs/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "build": "graph build",
+    "deploy": "graph deploy --node https://api.thegraph.com/deploy/ fritzschoff/subgraphs",
+    "create-local": "graph create --node http://localhost:8020/ fritzschoff/subgraphs",
+    "remove-local": "graph remove --node http://localhost:8020/ fritzschoff/subgraphs",
+    "deploy-local": "graph deploy --node http://localhost:8020/ --ipfs http://localhost:5001 fritzschoff/subgraphs",
+    "test": "graph test"
+  },
+  "dependencies": {
+    "@graphprotocol/graph-cli": "0.35.0",
+    "@graphprotocol/graph-ts": "0.28.1"
```
[the remark this answers is not in the document]
Comment (fritzschoff) [reply v3/subgraphs/package.json:16]: will do

File: v3/subgraphs/src/core.ts [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+        event.params.poolId.toString().concat('-').concat(event.params.collateralType.toHex())
+      )
+  );
+  if (position !== null) {
+    if (position.total_burned !== null) {
+      position.total_burned = position.total_burned!.plus(event.params.amount.toBigDecimal());
+    } else {
+      position.total_burned = event.params.amount.toBigDecimal();
+    }
+    if (position.total_burned && position.total_burned) {
```
Comment (bachstatter) [inline v3/subgraphs/src/core.ts:442]: This should be `position.total_burned && position.total_minted`

Comment (fritzschoff) [reply v3/subgraphs/src/core.ts:442]: Thanks but already fixed locally. It also changed a bit

File: .circleci/config.yml [hunk 7]
```diff
@@ -154,6 +154,8 @@ jobs:
       - store_artifacts:
           path: 'coverage'
 
+      - run: yarn workspace @snx-v3/subgraphs test
```
[the remark this answers is not in the document]
Comment (fritzschoff) [reply .circleci/config.yml:157]: seems pretty buggy. all the pool handlers are tested.

[the remark this answers is not in the document]
Comment (fritzschoff) [reply .circleci/config.yml:157]: Bildschirmfoto 2022-11-25 um 10 12 17
Bildschirmfoto 2022-11-25 um 10 12 28
Bildschirmfoto 2022-11-25 um 10 12 42
Bildschirmfoto 2022-11-25 um 10 12 52

[the remark this answers is not in the document]
Comment (fritzschoff) [reply .circleci/config.yml:157]: Bildschirmfoto 2022-11-25 um 10 17 07
you have to reexport the handlefunctions in the test file in order to make them visible for the coverage tool...

File: .eslintrc.js [hunk 8]
```diff
@@ -151,5 +151,22 @@ module.exports = {
...
+        i64: true,
+        assert: true,
+      },
+      rules: {
+        'prefer-const': 'off',
+        '@typescript-eslint/no-unused-vars': 'off',
+        '@typescript-eslint/ban-types': 'off',
+        '@typescript-eslint/no-array-constructor': 'off',
+      },
+    },
```
[the remark this answers is not in the document]
Comment (fritzschoff) [reply .eslintrc.js:170]: thank you!

**Comment by fritzschoff** [comment]
All the entities that are present are important according to @noahlitvin but I let him decide if we want to shrink the subgraph.
@noahlitvin please make the call

File: v3/subgraphs/tests/.latest.json [hunk 9]
```diff
@@ -0,0 +1,4 @@
+{
```
[the remark this answers is not in the document]
Comment (fritzschoff) [reply v3/subgraphs/tests/.latest.json:1]: will remove it all good

**Review by noahlitvin — APPROVED** [review]

**Pull request merged**